base/compiler/typeinfer.jl

```diff
@@ -95,26 +95,36 @@ const __measure_typeinf__ = RefValue{Bool}(false)
 function finish!(interp::AbstractInterpreter, caller::InferenceState;
                  can_discard_trees::Bool=may_discard_trees(interp))
     result = caller.result
-    valid_worlds = result.valid_worlds
-    if last(valid_worlds) >= get_world_counter()
-        # if we aren't cached, we don't need this edge
-        # but our caller might, so let's just make it anyways
-        store_backedges(result, caller.stmt_edges[1])
-    end
     opt = result.src
     if opt isa OptimizationState
-        result.src = opt = ir_to_codeinf!(opt)
+        result.src = ir_to_codeinf!(opt)
+    end
+    #@assert last(result.valid_worlds) <= get_world_counter() || isempty(caller.edges)
+    if isdefined(result, :ci_for_cache) || isdefined(result, :ci_as_edge)
```

Comment:
`isdefined(result, :ci_as_edge)` is actually dead, since we set the field after `typeinf` in the both places where we did it.